radio group should have a legend not a label

jelhan opened the pull request:
This changes the elements used for the label of a radio group from `<label>` to `<legend>`.
A `<label>` should only be used for the label of each option.

Simplified markup before this change:

```html
  <label>label of radio group</label>
  <div class="form-check">
    <input class="form-check-input" type="radio" name="gridRadios" id="gridRadios1" value="option1" checked>
    <label class="form-check-label" for="gridRadios1">
      First radio
    </label>
  </div>
  <div class="form-check">
    <input class="form-check-input" type="radio" name="gridRadios" id="gridRadios2" value="option2">
    <label class="form-check-label" for="gridRadios2">
      Second radio
    </label>
  </div>
```

After this change it looks like:

```html
  <legend>label of radio group</legend>
  <div class="form-check">
    <input class="form-check-input" type="radio" name="gridRadios" id="gridRadios1" value="option1" checked>
    <label class="form-check-label" for="gridRadios1">
      First radio
    </label>
  </div>
  <div class="form-check">
    <input class="form-check-input" type="radio" name="gridRadios" id="gridRadios2" value="option2">
    <label class="form-check-label" for="gridRadios2">
      Second radio
    </label>
  </div>
```

I would treat this as a bug fix even so it might break custom CSS styles. I think releasing as part of a minor version should be fine.

Belongs to #931

jelhan commented:
@simonihmig The tests are failing in Ember 3.4 and 2.18 cause of `@defaultValue` decorator. If I remove it from `@labelComponent` tests are passing. I've no idea whats going on here. Would appreciate if you could give me some hints what's going on here.

simonihmig commented:
@jelhan I had a look yesterday, but cannot tell from just looking at the code. Never seen that error before. Would have to reproduce locally and debug, it's weird...

simonihmig commented:
I tracked this down, seems to be an issue of `ember-decorators-polyfill`: https://github.com/pzuraq/ember-decorators-polyfill/issues/27

jelhan commented:
> I tracked this down, seems to be an issue of `ember-decorators-polyfill`: [pzuraq/ember-decorators-polyfill#27](https://github.com/pzuraq/ember-decorators-polyfill/issues/27)

Awesome. Thanks a lot for debugging this one!

simonihmig commented:
@jelhan v1.1.2 of ember-decorators-polyfill has been released with the fix that hopefully should fix the failing tests here. Could you update it here please and see if that is the case?

jelhan commented:
> @jelhan v1.1.2 of ember-decorators-polyfill has been released with the fix that hopefully should fix the failing tests here. Could you update it here please and see if that is the case?

Upgrade to `ember-decorators-polyfill@1.1.2` fixed the issues. Thanks a lot. Rebased to fix merge conflicts (`yarn.lock` only) afterwards.

Not sure if we should ship this as part of `3.x` or consider as breaking change and wait until `4.x`. As 4.x is under active development that would be fine for me. What do you prefer?

simonihmig commented:
We could "sell" it as a bug fix, but it still could feel like a breaking change (CSS selectors not applying anymore, which no test suite will detect, unless using visual regression testing), so I think shipping that with 4.x would make sense!

jelhan commented:
Rebased on top of `dev-v4` and changed target branch. Should be ready to merge if CI passes.

Needed to close and reopen cause Travis was not picking up the commit after last rebase.

simonihmig commented:
The failing tests seem to be unrelated. Also saw these on master for ember-canary... 🤔

jelhan commented:
> The failing tests seem to be unrelated. Also saw these on master for ember-canary... thinking

Have seen unstable canary for some addons I was working on the last days. Don't think this is related.
